callstats-java-sdk/src/main/java/io/callstats/sdk/CallStats.java

```diff
@@ -0,0 +1,227 @@
+package io.callstats.sdk;
+
+import io.callstats.sdk.messages.AuthorizeRequestMessage;
+import io.callstats.sdk.messages.AuthorizeResponseMessage;
+import io.callstats.sdk.messages.ChallengeRequestMessage;
+import io.callstats.sdk.messages.ChallengeResponseMessage;
+
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.HttpResponse;
+import org.apache.http.ParseException;
+import org.apache.http.util.EntityUtils;
+
+import com.google.gson.Gson;
+
+/**
+ *
+ * @author Karthik Budigere
+ */
+public class CallStats {
+	
+	private static final String CS_VERSION = "0.1.0";
+	private static final int RESPONSE_STATUS_SUCCESS = 200;
+	private static final int SERVER_ERROR = 500;
+	private static final int INVALID_PROTO_FORMAT_ERROR = 400;
+	private static final int INVALID_PARAM_ERROR = 403;
+	private static final int GATEWAY_ERROR = 502;
+	
+	private static final String END_POINT_TYPE = "VideoBridge";
+	
+	private CallStatsHttpClient httpClient;
+	private int appId;
+	private String appSecret;
+	private String bridgeId;
+	private Gson gson;
+	private CallStatsInitListener listener;
+	private int  authenticationRetryTimeout = 5000;
+	private static final ScheduledExecutorService scheduler = 
+			  Executors.newSingleThreadScheduledExecutor();
+	
+	public CallStatsHttpClient getHttpClient() {
+		return httpClient;
+	}
+
+	public void setHttpClient(CallStatsHttpClient httpClient) {
+		this.httpClient = httpClient;
+	}
+
+	public int getAppId() {
+		return appId;
+	}
+
+	public void setAppId(int appId) {
+		this.appId = appId;
+	}
+
+	public String getAppSecret() {
+		return appSecret;
+	}
+
+	public void setAppSecret(String appSecret) {
+		this.appSecret = appSecret;
+	}
+
+	public String getBridgeId() {
+		return bridgeId;
+	}
+
+	public void setBridgeId(String bridgeId) {
+		this.bridgeId = bridgeId;
+	}
+
+	public CallStats() {
+		gson = new Gson();
+	}
+
+	public void intialize(int appId, String appSecret, String bridgeId,CallStatsInitListener listener) {
+		if(appId <= 0 || appSecret == null || bridgeId == null) {
+			throw new IllegalArgumentException("intialize: Arguments cannot be null");
+		}
+		
+		this.appId = appId;
+		this.appSecret = appSecret;
+		this.bridgeId = bridgeId;
+		this.listener = listener;
+		
+		httpClient = new CallStatsHttpClient();
+		sendAsyncAuthenticationRequest(appId,appSecret,bridgeId);
+	}
+	
+	private String getHmacResponse(String challenge) {
```

Comment:
Don't you want to put whole authentication functionality as a separate class which CallStats object treats as "black box"?